lib/flow_runner/spec/block.ex

```diff
@@ -222,62 +215,56 @@ defmodule FlowRunner.Spec.Block do
       {:ok, context, block}
     else
       {:invalid, reason} ->
         Logger.info("Fetching default block because #{reason}")
-        {:ok, context} = Block.evaluate_user_input(block, context, user_input)
-        Block.fetch_default_block(block, flow, context)
 
-      err ->
-        err
+        {:ok, context} = evaluate_user_input(block, context, user_input)
+
+        fetch_default_block(block, flow, context)
     end
   end
 
+  @spec fetch_default_block(Block.t(), Flow.t(), Context.t()) ::
+          {:error, iodata} | {:ok, Context.t(), Block.t() | nil}
   def fetch_default_block(block, %Flow{} = flow, %Context{} = context) do
-    {:ok, %Exit{destination_block: destination_block}} =
-      Block.evaluate_default_exit(block, context)
-
-    if destination_block == "" || destination_block == nil do
-      {:ok, %Context{context | finished: true}, nil}
-    else
-      case Flow.fetch_block(flow, destination_block) do
-        {:ok, next_block} -> {:ok, context, next_block}
-        {:error, reason} -> {:error, reason}
-      end
+    case evaluate_default_exit(block) do
+      {:ok, %Exit{destination_block: destination_block}}
+      when is_nil(destination_block) or destination_block == "" ->
+        {:ok, %Context{context | finished: true}, nil}
+
+      {:ok, %Exit{destination_block: destination_block}} ->
+        with {:ok, next_block} <- Flow.fetch_block(flow, destination_block) do
+          {:ok, context, next_block}
+        end
+
+      {:error, reason} ->
+        {:error, reason}
     end
   end
 
-  @spec fetch_next_block(
-          Block.t(),
-          Flow.t(),
-          Context.t()
-        ) ::
-          {:error, iodata}
-          | {:ok, Context.t(), Block.t()}
+  @spec fetch_next_block(Block.t(), Flow.t(), Context.t()) ::
+          {:error, iodata} | {:ok, Context.t(), Block.t()}
   def fetch_next_block(block, %Flow{} = flow, %Context{} = context) do
-    {:ok, %Exit{destination_block: destination_block}} = Block.evaluate_exits(block, context)
-
-    if destination_block == "" || destination_block == nil do
-      {:ok, %Context{context | finished: true}, nil}
-    else
-      case Flow.fetch_block(flow, destination_block) do
-        {:ok, next_block} -> {:ok, context, next_block}
-        {:error, reason} -> {:error, reason}
-      end
+    case evaluate_exits(block, context) do
+      {:ok, %Exit{destination_block: destination_block}}
+      when is_nil(destination_block) or destination_block == "" ->
+        {:ok, %Context{context | finished: true}, nil}
+
+      {:ok, %Exit{destination_block: destination_block}} ->
+        with {:ok, next_block} <- Flow.fetch_block(flow, destination_block) do
+          {:ok, context, next_block}
+        end
+
+      {:error, reason} ->
+        {:error, reason}
     end
   end
 
-  @spec evaluate_exits(Block.t(), Context.t()) ::
-          {:ok, Exit.t()} | {:error, any()}
+  @spec evaluate_exits(Block.t(), Context.t()) :: {:ok, Exit.t()} | {:error, any()}
```

Comment:
It can only return this error: https://github.com/turnhub/flow_runner/blob/9752d7db571618b643b126df9fc2613320fdfdf4/lib/flow_runner/spec/block.ex#L293-L293
```suggestion
  @spec evaluate_exits(Block.t(), Context.t()) :: {:ok, Exit.t()} | {:error, String.t}
```

Reply:
Good catch @joaothallis! I have applied it in [985cbbf](https://github.com/turnhub/flow_runner/pull/258/commits/985cbbf125300890a5652327defc6b940ca07b98).